include/crm/common/strings_internal.h

```diff
@@ -142,6 +142,7 @@ bool pcmk__char_in_any_str(int ch, ...) G_GNUC_NULL_TERMINATED;
 int pcmk__strcmp(const char *s1, const char *s2, uint32_t flags);
 int pcmk__numeric_strcasecmp(const char *s1, const char *s2);
 void pcmk__str_update(char **str, const char *value);
+void pcmk__g_strcat(GString *buffer, ...);
```

Comment:
If you add G_GNUC_NULL_TERMINATED to the function prototype, gcc will be able to warn on call sites that don't include a NULL at the end.  See pcmk__strcase_any_of, for instance.

Reply:
Oh neat. You know all the compiler tricks :)

Reply:
My answer to everything is "I bet the compiler can do this for me..."